### DISCUSSION TITLE
Add a wallet selector and add support for Talisman wallet

### DISCUSSION THREAD

**Discussion by itsbirdo** [darwinia-network/darwinia#837] [Ideas]
### **Problem?**
When trying to connect the Talisman (https://talisman.xyz/) wallet, it's not officially supported and while it DOES work (you can connect your wallet and sign TXs), it's not the smoothest experience. 

### **Solution**
Talisman have created an easy library for dapps to implement (See below). If you want to use your own UI, you can use just part of the package (to do most of the hardwork for you!) and add your own UI on top. Or you can use our built in UI to make the process easier.
https://github.com/TalismanSociety/talisman-connect

As more wallets come to the dotsama ecosystem, dapps will need a better way to manage them, this feature request provides some future proofing and also let's your dapps support Talisman!
You can see a working implementation of this at -> https://app.talisman.xyz/

### **User journey**
As a user on your dapp
Given I click, Connect
Then I want to see a modal display which allows me to select my preferred wallet

**first time user**
When I select Talisman
Then I want to see a dialog appear showing the different wallets I could link (this is handled by the extension)
And after selecting one or more wallets, the modal should disappear and the dapp should re-load with my data

**Returning user**
Then when I click the wallet of choice, and I select Talisman
And after selecting one or more wallets, the modal should disappear and the dapp should re-load with my data

### **Example screenshots**
![image](https://user-images.githubusercontent.com/93915849/157373170-0e424387-d80e-43c9-8a89-67bf5b54450e.png)
![image](https://user-images.githubusercontent.com/93915849/157373176-2691bb02-b96d-4cca-a25b-3b4b8b58e601.png)

**Comment by 1022ranji** [comment]
Hi, glad to see you here. we'll check the code first, and then discuss the integration. But because there are some other developed tasks that need to do, this might cost a long time.

**Reply by itsbirdo** [reply]
@1022ranji sounds good - looking forward to an update!

**Reply by meganskye** [reply]
> Hi, glad to see you here. we'll check the code first, and then discuss the integration. But because there are some other developed tasks that need to do, this might cost a long time.

Hi Ran JI, SubWallet team has posted a new thread in our discussion group with a link to Subconnect, which i think will make integration of Talisman wallet easier. See quote below for reference: 

"refer to the SubConnect module, we built a Mini app to connect all wallets in the Polkadot ecosystem which includes Polkadot{.js}, SubWallet, and Talisman.

    Demo link: https://connect.subwallet.app/
    Source Code: https://github.com/Koniverse/SubConnect
    Video Demo how it works:

https://www.loom.com/share/3248348dafef4b28aee365b90e1e0b16

Currently, I have not seen a library that fully supports wallets in the Dotsama ecosystem like SubConnect. Projects seem a bit confused to integrate multiple wallets, that's why we built it"

**Reply by itsbirdo** [reply]
> > Hi, glad to see you here. we'll check the code first, and then discuss the integration. But because there are some other developed tasks that need to do, this might cost a long time.
> 
> Hi Ran JI, SubWallet team has posted a new thread in our discussion group with a link to Subconnect, which i think will make integration of Talisman wallet easier. See quote below for reference:
> 
> "refer to the SubConnect module, we built a Mini app to connect all wallets in the Polkadot ecosystem which includes Polkadot{.js}, SubWallet, and Talisman.
> 
> ```
> Demo link: https://connect.subwallet.app/
> Source Code: https://github.com/Koniverse/SubConnect
> Video Demo how it works:
> ```
> 
> https://www.loom.com/share/3248348dafef4b28aee365b90e1e0b16
> 
> Currently, I have not seen a library that fully supports wallets in the Dotsama ecosystem like SubConnect. Projects seem a bit confused to integrate multiple wallets, that's why we built it"

Hey @meganskye fyi that is a clone of our Talisman connect repo w/ Subwallet added in. Easy enough to use the Talisman connect library as originally proposed and add Subwallet as an option :)

**Reply by meganskye** [reply]
whoopsie i was hoping the suggestion would help expedite the integration from our side but i guess not. thank you for clearing that up

**Comment by itsbirdo** [comment]
Hey all - FYI we've pushed some changes to the repo to:
1. add subwallet as an option
2. make the integration even easier to do - one of the devs at a hackathon yesterday integrated it in less than 5 minutes.

Would love to get this sorted soon!

**Comment by hackfisher** [comment]
Talisman is well supported, and most the DApps in Darwinia chain not can choose lots of wallet using kits like https://www.rainbowkit.com/
